### ISSUE TITLE
Provide image description

### ISSUE THREAD

**Issue by asbjornu** [GitTools/build-images#35]
Currently there's no description `LABEL` added to our Docker images. We should probably provide one.

<img width="747" alt="No description provided" src="https://github.com/GitTools/build-images/assets/12283/dc2a6539-48b4-44b1-a040-e20c7548b865">

**Comment by asbjornu** [comment]
Right. Is it possible to interpolate the environment, architecture, OS version, etc., into the descriptions?

**Comment by arturcic** [comment]
This are the LABELs we can use
https://github.com/opencontainers/image-spec/blob/main/annotations.md. 
> Is it possible to interpolate the environment, architecture, OS version, etc., into the descriptions?

I think that is doable, as we are already append programatically the dockerfile with additional content

**Comment by asbjornu** [comment]
How does the `deps`, `build-images` and `gitversion` images differ? As in, how are they used? Only the `gitversion` images are consumed directly, no?

**Comment by arturcic** [comment]
`build-images` I think are also used by libgit2sharp to validate their build, at least that is what I contributed with there

**Comment by asbjornu** [comment]
This is what I think we should annotate the GitVersion Docker images with:

| `LABEL`                                  | Value
| :--------------------------------------- | :---
| `org.opencontainers.image.created`       | `2023-12-29T13:35:12Z`
| `org.opencontainers.image.authors`       | `GitVersion Maintainers`
| `org.opencontainers.image.url`           | The URL to the published image in the appropriate registry.
| `org.opencontainers.image.documentation` | https://gitversion.net/docs/usage/docker
| `org.opencontainers.image.source`        | https://github.com/GitTools/GitVersion/blob/main/build/docker/Dockerfile
| `org.opencontainers.image.version`       | The same version as that of the packaged GitVersion binary.
| `org.opencontainers.image.revision`      | SHA of the GitVersion commit that was tagged and built.
| `org.opencontainers.image.vendor`        | `GitTools`
| `org.opencontainers.image.licenses`      | `MIT`
| `org.opencontainers.image.ref.name`      | `6.0.0-beta.3-fedora.36-6.0-arm64`
| `org.opencontainers.image.title`         | `GitVersion`
| `org.opencontainers.image.description`   | `From git log to SemVer in no time`
| `org.opencontainers.image.base.digest`   | Seems complicated to generate, so we can probably just punt on this one.
| `org.opencontainers.image.base.name`     | `$REGISTRY/gittools/build-images:$DISTRO-sdk-$DOTNET_VERSION`

I'm not sure what should be done for `deps` and `build-images`, but many of the values can most likely be reused.

**Comment by asbjornu** [comment]
Useful read: https://adrianmouat.com/posts/annotations-and-labels-in-container-images/

**Comment by arturcic** [comment]
> This is what I think we should annotate the GitVersion Docker images with:
> 
> | `LABEL`                                  | Value
> | :--------------------------------------- | :---
> | `org.opencontainers.image.created`       | `2023-12-29T13:35:12Z`
> | `org.opencontainers.image.authors`       | `GitVersion Maintainers`
> | `org.opencontainers.image.url`           | The URL to the published image in the appropriate registry.
> | `org.opencontainers.image.documentation` | `https://gitversion.net/docs/usage/docker`  (once https://github.com/GitTools/GitVersion/pull/3858 is merged)
> | `org.opencontainers.image.source`        | `https://github.com/GitTools/GitVersion/blob/main/build/docker/Dockerfile`
> | `org.opencontainers.image.version`       | The same version as that of the packaged GitVersion binary.
> | `org.opencontainers.image.revision`      | SHA of the GitVersion commit that was tagged and built.
> | `org.opencontainers.image.vendor`        | `GitTools`
> | `org.opencontainers.image.licenses`      | `MIT`
> | `org.opencontainers.image.ref.name`      | `6.0.0-beta.3-fedora.36-6.0-arm64`
> | `org.opencontainers.image.title`         | `GitVersion`
> | `org.opencontainers.image.description`   | `From git log to SemVer in no time`
> | `org.opencontainers.image.base.digest`   | Seems complicated to generate, so we can probably just punt on this one.
> | `org.opencontainers.image.base.name`     | `$REGISTRY/gittools/build-images:$DISTRO-sdk-$DOTNET_VERSION`
> 
> I'm not sure what should be done for `deps` and `build-images`, but many of the values can most likely be reused.

I will have a look and update the labels and I will check about annotations, but that will be after my vacation
